# Axe 1 : Conditions de travail

Dans ce notebook, j'étudie les variables liées aux facteurs organisationnelles des employés afin d'évaluer leur impact sur l'attrition.

Ce notebook a pour objectif :
- d'explorer les variables de l'axe,
- de comparer avec l'attrition
- d'explorer des relations internes à l’axe
- de visualiser
- retenir seulement ce qui ressort clairement
---

## 1. Exploration des variables de l’axe
---

### 1.1 Import des librairies & chargement des données
---

In [ ]:
# import librairies 

import pandas as pd
import pyarrow
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [ ]:
# chargement du df
df_clean = pd.read_parquet("C:/Users/Kemu/Documents/Formation/Projet Pro Data/2_Projets_Tests/projet-RH_test/data/processed/employees_clean.parquet")
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Age                       1470 non-null   int64 
 1   Attrition                 1470 non-null   string
 2   BusinessTravel            1470 non-null   string
 3   DailyRate                 1470 non-null   int64 
 4   Department                1470 non-null   string
 5   DistanceFromHome          1470 non-null   int64 
 6   Education                 1470 non-null   int64 
 7   EducationField            1470 non-null   string
 8   EmployeeCount             1470 non-null   int64 
 9   EmployeeNumber            1470 non-null   int64 
 10  EnvironmentSatisfaction   1470 non-null   int64 
 11  Gender                    1470 non-null   string
 12  HourlyRate                1470 non-null   int64 
 13  JobInvolvement            1470 non-null   int64 
 14  JobLevel                

### 1.2 Creation du DataFrame Condition de travail
---
#### Colonnes incluses dans l'axe "Condition de travail"

Cet axe regroupe les caractéristiques individuelles :
- BusinessTravel = Structure du poste = leger
- Department = Structure du poste = leger mais si gros impact interresant pour l'attrition
- JobLevel = Structure du poste = leger
- JobRole = Structure du poste = detaillé
- MonthlyIncome = Rémunération = detaillé
- OverTime = structure poste = detaillé
- PercentSalaryHike = Rémuneration = leger
- StockOptionLevel = Rémuneration = leger
- TotalWorkingYears = structure poste = a voir si impact
- TrainingTimesLastYear = evolution = 
- YearsAtCompany = ancienneté = detaillé
- YearsInCurrentRole = carriere = detaillé
- YearsSinceLastPromotion = evolution = detaillé
- YearsWithCurrManager = manager = a voir en lien avec departement
 
donnant un contexte organisationnelle pouvant influencer l'attrition.

In [ ]:
# colonnes condition de travail
col_trav = ['BusinessTravel', 'Department','JobLevel',"JobRole",'MonthlyIncome','OverTime','PercentSalaryHike','StockOptionLevel',
             "TotalWorkingYears", "TrainingTimesLastYear", "YearsAtCompany","YearsInCurrentRole","YearsSinceLastPromotion","YearsWithCurrManager" ]

# creation df_condition_Trav
df_condit_Trav = df_clean[col_trav + ['Attrition']].copy()
df_condit_Trav.head()

#### 1.2.1 Création des tranches 



#### 1.2.2 Regroupement distance domicile–travail



#### 1.2.3 Renommage 



#### 1.2.4 Renommage 


#### 1.2.5 Regroupement 


### 1.3  Analyse individuelle des variables 
---
Dans cette section, j’analyse plusieurs caractéristiques des conditions de travails des employés afin de mieux comprendre le contexte organisationnelle de la population étudiée.  
Les variables sont analysées par sous themes :
- structure du poste
- rémunération
- ancienneté / carrière
- management / évolution
Certaine variables seront plus appronfondis que d'autresà , cette analyses se fera à partir de leur fréquence, à l’aide de tableaux de proportions et de visualisations graphiques.


#### 1.3.1 Structure du poste
---


##### Tableau des proportions

##### Histogramme des tranches d'âge




#### 1.3.2 Rémuneration
---



##### Tableau des proportions

#### 

#### Répartition des employés par 


#### 1.3.3 Ancienneté et carriere
---

##### Tableau des proportions

##### Répartition des employés 


#### 1.3.4 Management et évolution
---

##### Tableau des proportions

##### Repartitions des employes par 


## 2. Attrition — Comparaisons simples
---

### 2.1 Introduction
---
Dans cette section, je compare la variable cible, l’attrition, aux variables des conditions de travails analysées précédemment, afin d’identifier d’éventuelles relations entre ces caractéristiques et le départ des employés.


##### Tableau de proportions Attrition Yes / No

In [ ]:
attrition = df_vie_perso['Attrition'].value_counts(normalize=True).sort_values(ascending=False).round(3)*100

df_attrition = attrition.reset_index()
df_attrition.columns = ['Attrition', "Pourcentage"]

df_attrition

,Attrition,Pourcentage
0,No,83.9
1,Yes,16.1


D’après ce tableau, le taux d’attrition global observé dans le jeu de données est d’environ **16 %**, ce qui constitue le niveau de référence pour les analyses suivantes.
Dans les analyses suivantes, l’accent est mis sur le taux d’attrition (Attrition = Yes), les non-départs représentant le complément à 100 %.

### 2.2 Analyse de l’attrition selon les variables des conditions de travail
---
Les variables sont comparées à partir de leur fréquence, à l’aide de tableaux de proportions et de visualisations graphiques.  
Cette approche permet d’identifier des différences de comportement entre les groupes avant toute analyse approfondie.

#### 2.2.1 Attrition et Department
---

##### Tableau de proportions ­­

##### Graphique — Analyse de la relation entre 

##### Observation descriptive

#### 2.2.2 Attrition et JobRole
---


##### Tableau de proportions

##### Graphique — Analyse de la relation 

##### Observation descriptive

#### 2.2.3 Attrition et MonthlyIncome
---


##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive

#### 2.2.4 Attrition et overtime
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive

#### 2.2.5 Attrition et TotalWorkingYears
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

#### 2.2.6 Attrition et TrainingTimesLastYear
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

#### 2.2.7 Attrition et YearsAtCompany
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

#### 2.2.8 Attrition et YearsInCurrentRole
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

#### 2.2.9 Attrition et YearsSinceLastPromotion
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

#### 2.2.14 Attrition et YearsWithCurrManager
---

##### Tableau de proportions

##### Graphique — Analyse de la relation entre 

##### Observation descriptive 

Une fois les comparaisons simples réalisées, j’explore les relations internes à l’axe afin d’identifier d’éventuels patterns ou corrélations.

## 3. Relations internes à l’axe condition de travail
----

### 3.1 Introduction

Après avoir analysé les relations entre les variables personnelles et l’attrition, cette section vise à explorer les relations internes entre les variables de l’axe *Condition de travail*, afin d’identifier d’éventuels patterns ou profils récurrents.  
Les variables sont analysées à partir de tableaux croisés et de visualisations graphiques descriptives.

---


### 3.2 Analyse des relations entre variables organisationnelles

#### 3.2.1 Relation entre 
---

##### Tableaux croisés

##### Visualisation

##### Observations descriptives



#### 3.2.2 Relation entre 
---


##### Tableaux croisés

##### Visualisation

##### Observations descriptives


#### 3.2.3 Relation entre 
---


##### Tableau croisé

##### Visualisation

##### Observations descriptives



Après avoir analysé ces trois relations, cette section synthétise les principaux patterns observés.

### 3.3 Synthèse des patterns observés
---
Les analyses des relations internes à l’axe *Condition de travail* mettent en évidence plusieurs patterns structurants :

- **Pattern 1 — Relation entre**  
  
- **Pattern 2 — Relation entre**  
  
- **Pattern 3 — Relation entre**  
  
  Ces patterns permettent de mieux contextualiser les résultats observés précédemment et servent de base à la synthèse globale de l’axe Condition de travail.
  ---

# 4. Résultats clés 
---


---

# 5. Conclusion de l’axe “Condition de travail”
---

